algorithms/polish.sbl

```diff
@@ -0,0 +1,229 @@
+stringescapes {}
+
+stringdef a' '{U+0105}' 
+stringdef e' '{U+0119}' 
+stringdef l' '{U+0142}' 
+stringdef s' '{U+015b}' 
+stringdef c' '{U+0107}' 
+stringdef o' '{U+00f3}' 
+
+
+
+
+externals (stem)
+integers (pos)
+
+
+routines (remove_nouns remove_dimunitive remove_adjective_ends remove_verbs_ends remove_adverbs_ends 
+	remove_plural_forms remove_general_ends)
+
+
+define remove_nouns as (
+			($(len > 7)
+			test ($pos = (len - 5)
+			hop pos
+			('zacja' or 'zacja' or 'zacji'))
+			($pos = (len - 4)
+			hop pos
+			([tolimit] delete))) or
+			($(len > 6)
+			test($pos = (len - 4)
+			hop pos
+			('acja' or 'acji' or 'acj{a'}' or 'tach' or 'anie' or 'enie' or 'eniu' or 'aniu'))
+			($pos = (len -4)
+			hop pos
+			([tolimit] delete))) or
+			($(len > 6)
+			test($pos = (len - 4)
+			hop pos
+			('tyka'))
+			($pos = (len - 2)
+			hop pos
+			([tolimit] delete))
+			) or
+			($(len > 5)
+			test ($pos = (len - 3)
+			hop pos
+			('ach' or 'ami' or 'nia' or 'niu' or 'cia' or 'ciu'))
+			($pos = (len -3)
+			hop pos
+			([tolimit] delete))
+			) or
+			($(len > 5)
+			test ($pos = (len - 3)
+			hop pos
+			('cji' or 'cja' or 'cj{a'}'))
+			($pos = (len - 2)
+			hop pos
+			([tolimit] delete ))) or
+			($(len > 5)
+			test($pos = (len - 2)
+			hop pos
+			('ce' or 'ta'))
+			($pos = (len - 2)
+			hop pos
+			([tolimit] delete))
+			)
+
+)
+
+define remove_dimunitive as (
+		($(len > 6)
+		test ($pos = (len - 5)
+		hop pos
+		('eczek'  or 'iczek' or 'iszek' or 'aszek' or 'uszek')) 
+		($pos = (len - 5)
+		hop pos
+		([tolimit] delete))) or
+		($(len > 6)
+		test ($pos = (len - 4)
+		hop pos
+		('enek' or 'ejek' or 'erek'))
+		($pos = (len - 2)
+		hop pos
+		([tolimit] delete))) or
+		($(len > 4)
+		test ($pos = (len - 2)
+		hop pos
+		('ek' or 'ak'))
+		($pos = (len -2)
+		hop pos
+		([tolimit] delete))
+		)
```

Comment:
This would benefit from using snowball's `among` feature.  This can be used to remove suffixes from the string without having to do all this tedious calculating of offsets, and that would fix a lot of the current problems when working in UTF-8.